test/unit/lualib/lualib.spec.ts

```diff
@@ -21,6 +21,23 @@ export class LuaLibTests
         Expect(result).toBe(JSON.stringify(expected));
     }
 
+    @TestCase([0, 2, 4, 8], 10, -1)
+    @TestCase([0, 2, 4, 8], 4, 2)
+    @TestCase([0, 2, 4, 8], 8, 3)
+    @Test("findIndex")
+    public findIndex(inp: number[], searchEl: number, expected: number): void
+    {
+        const result = util.transpileAndExecute(
+            `let arrTest = ${JSON.stringify(inp)};
+            return JSONStringify(arrTest.findIndex((elem, index) => {
```

Comment:
Would be nice if we could also have a test checking if the index and array parameters of the callback are set correctly by the lualib function.

Reply:
I've updated, let me know what you think